src/joins.js

```diff
@@ -60,29 +89,79 @@ const getFieldName = (item, tableName) => {
 };
 
 const extendSingular = (args) => {
-  const { items, mergeTableName, mergeItems, mergeItemLeading, spread } = args;
+  const { items, mergeTableName, mergeItems, spread } = args;
+  return items.map((item) => {
+    const relevantItem = mergeItems.find((mergeItem) =>
+      findItem(item, mergeItem, args)
+    );
+    const toAdd = spread ? relevantItem : { [mergeTableName]: relevantItem };
+    return { ...item, ...toAdd };
+  });
+};
+
+const extendMulti = (args) => {
+  let { items, mergeTableName, mergeItems } = args;
+  mergeTableName = pluralize.plural(mergeTableName);
+  return items.map((item) => {
+    const relevantItems = mergeItems.filter((mergeItem) =>
+      findItem(item, mergeItem, args)
+    );
+    return { ...item, [mergeTableName]: relevantItems };
+  });
+};
+
```

Comment:
Ik denk dat we `extendMulti` en `through` toch ook wel samen kunnen voegen. Het enige verschil lijkt de fn die je aanroept (`findItem` vs `findItemThrough`) en de `isBaseMultiple` check. De fn kan je meegeven als arg en `isBaseMultiple` wordt alleen gebruikt om de tableName te setten, dus dat kan opgelost worden door de tableName al juist mee te geven aan de fn.